### PR TITLE
fix issue deleting proxy from storage tables

### PR DESCRIPTION
**Pull request by bmc-msft**
This issue was introduced in 2.18.0.

### CODE REVIEW

File: src/api-service/__app__/onefuzzlib/proxy.py [hunk 1]
```diff
@@ -165,7 +165,10 @@ def stopping(self) -> None:
             self.stopped()
 
     def stopped(self) -> None:
+        if self.state != VmState.stopped:
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/proxy.py:168]: Should `set_state()` be responsible for this "set to `<state>` if not `<state>`" logic? Do we ever _not_ want this?

File: src/api-service/__app__/onefuzzlib/proxy.py [hunk 2]
```diff
@@ -165,7 +165,10 @@ def stopping(self) -> None:
...
+            self.set_state(VmState.stopped)
         logging.info(PROXY_LOG_PREFIX + "removing proxy: %s", self.region)
+        send_event(EventProxyDeleted(region=self.region, proxy_id=self.proxy_id))
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/proxy.py:171]: Should we only send this _after_ successfully calling `delete()`?

File: src/api-service/__app__/onefuzzlib/proxy.py (hunk 1 above)
Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/proxy.py]: This is already done in set_state, so I removed the additional check here.

File: src/api-service/__app__/onefuzzlib/proxy.py (hunk 2 above)
Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/proxy.py:170]: I think it's a toss-up.  I was trying to remove the conceptual use-after-delete.

Comment (ranweiler) [reply src/api-service/__app__/onefuzzlib/proxy.py:170]: That's legit. This and the other (related) comments about events / states are more discussion points. We can revisit the big picture going forward; I think the only thing that matters is that we have a well defined point in the business logic at which we expect to fire a given event!

**Review by ranweiler — APPROVED** [review]

**Pull request merged**